nestor_api/lib/config.py

```diff
@@ -125,3 +126,25 @@ def __resolve(value, variables, path, errors):
         raise AggregatedConfigurationError(errors)
 
     return resolved_config
+
+
+def get_previous_step(config_object: dict, target: str) -> Optional[str]:
+    """ Returns the previous step in the defined workflow """
+    index = config_object["workflow"].index(target)
+    if index > 0:
+        return config_object["workflow"][index - 1]
+    return None
+
+
+# pylint: disable=bad-continuation
+def list_apps_config(
+    config_path=Configuration.get_config_path(), apps_path=Configuration.get_config_app_folder()
+) -> dict:
+    """Retrieves all of the application names."""
+    apps_path = os.path.join(config_path, apps_path)
+
+    if not os.path.isdir(apps_path):
+        raise ValueError(apps_path)
+    apps_files = [f for f in os.listdir(apps_path) if os.path.isfile(os.path.join(apps_path, f))]
```

Comment:
Are you sure that `listdir` will not throw itself an Exception here?

Reply:
Outdated as I changed a little bit the way it is handled.